### ISSUE TITLE
Build error[E0282]: type annotations needed

### ISSUE THREAD

**Issue by fullmetalScience** [comit-network/xmr-btc-swap#1459]
With *master* at commit db400ad4c4bf4c3f5e252966ff4fda8614242c2e:

```
$ cargo build
warning: virtual workspace defaulting to `resolver = "1"` despite one or more workspace members being on edition 2021 which implies `resolver = "2"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
note: for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
   Compiling libgit2-sys v0.14.2+1.5.1
   Compiling libsqlite3-sys v0.24.2
   Compiling secp256kfun v0.7.1 (https://github.com/LLFourn/secp256kfun#9657d8c1)
   Compiling electrum-client v0.12.0
error[E0282]: type annotations needed for `point::Point<point_type::EvenY, S>`
   --> ~/.cargo/git/checkouts/secp256kfun-fb74af16957b2754/9657d8c/secp256kfun/src/point.rs:496:45
    |
496 |         Point::from_xonly_bytes(bytes).map(|p| p.set_secrecy::<S>())
    |                                             ^
    |
help: consider giving this closure parameter an explicit type, where the type for type parameter `S` is specified
    |
496 |         Point::from_xonly_bytes(bytes).map(|p: point::Point<point_type::EvenY, S>| p.set_secrecy::<S>())
    |                                              ++++++++++++++++++++++++++++++++++++

   Compiling git2 v0.16.1
   Compiling sqlx-core v0.6.3
For more information about this error, try `rustc --explain E0282`.
error: could not compile `secp256kfun` (lib) due to previous error
```

**Comment by delta1** [comment]
rust version?

**Comment by fullmetalScience** [comment]
`rustc 1.74.0`

**Comment by delta1** [comment]
thanks will take a look. in the meanwhile 1.70 should work (it does in the CI)

**Comment by ikmckenz** [comment]
Also broken on 1.72

**Comment by delta1** [comment]
resolved short term by using rust 1.70 and long term by #1466
